packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -397,86 +399,100 @@ export function ChatSettings(props: ChatSettingsProps): JSX.Element {
+            label="Inline completion model"
+            disabled={!isCompleterEnabled}
+            onChange={e => {
+              const clmGid = e.target.value === 'null' ? null : e.target.value;
+              if (clmGid === null) {
+                setClmProvider(null);
+                return;
+              }
+
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              const nextClmProvider = getProvider(clmGid, server.lmProviders)!;
+              // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              const nextClmLocalId = getModelLocalId(clmGid)!;
+
+              setClmProvider(nextClmProvider);
+              setCompletionHelpMarkdown(nextClmProvider?.help ?? null);
+              if (nextClmProvider.registry) {
+                setClmLocalId('');
+                setShowClmLocalId(true);
+              } else {
+                setClmLocalId(nextClmLocalId);
+                setShowClmLocalId(false);
+              }
+            }}
+            MenuProps={{ sx: { maxHeight: '50%', minHeight: 400 } }}
+          >
+            <MenuItem value="null">None</MenuItem>
+            {server.lmProviders.providers.map(lmp =>
+              lmp.models
+                .filter(lm => lmp.completion_models.includes(lm))
+                .map(lm => (
+                  <MenuItem value={`${lmp.id}:${lm}`}>
+                    {lmp.name} :: {lm}
+                  </MenuItem>
+                ))
+            )}
+          </Select>
+          {showClmLocalId && (
+            <TextField
+              label={clmProvider?.model_id_label || 'Local model ID'}
+              value={clmLocalId}
+              onChange={e => setClmLocalId(e.target.value)}
+              fullWidth
+            />
+          )}
+          {completionHelpMarkdown && (
+            <RendermimeMarkdown
+              rmRegistry={props.rmRegistry}
+              markdownStr={completionHelpMarkdown}
+              complete
+            />
+          )}
+          {clmGlobalId && (
+            <ModelFields
+              fields={clmProvider?.fields}
+              values={fields}
+              onChange={setFields}
+            />
+          )}
+        </Box>
+      ) : (
+        <p>No Inline Completion models.</p>
       )}
 
       {/* API Keys section */}
       <h2 className="jp-ai-ChatSettings-header">API Keys</h2>
       {/* API key inputs for newly-used providers */}
-      {Object.entries(apiKeys).map(([apiKeyName, apiKeyValue], idx) => (
-        <TextField
-          key={idx}
-          label={apiKeyName}
-          value={apiKeyValue}
-          fullWidth
-          type="password"
-          onChange={e =>
-            setApiKeys(apiKeys => ({
-              ...apiKeys,
-              [apiKeyName]: e.target.value
-            }))
-          }
-        />
-      ))}
+      {Object.entries(apiKeys).length === 0 ? (
+        <p>No API Keys needed for selected model.</p>
```

Comment:
Note: this is also shown when no model is selected. Not sure if we need to make this conditional on selection.

Reply:
I think that makes sense and is acceptable UX; if there is no selected model, then clearly no API keys are needed.